packages/SwingSet/docs/vat-upgrade.md

```diff
@@ -43,30 +43,36 @@ As vats run, they send and receive messages. When these messages carry object re
 
 Each export represents an obligation. Other vats might send a message to the exported object, or they might send a message that references the exported object, and we must be prepared to handle it. So the primary job of the v2 vat code is to satisfy all the obligations incurred by v1. Another way to express this is that the kernel c-list, which maps kernel "krefs" to vat "vrefs" for each object, must be satisfied: every vref must be reassociated somehow, or marked as broken.
 
-There are two basic kinds of exports:
+There are three basic categories of exports (cf. [A Taxonomy of Exo-making Functions](https://github.com/endojs/endo/blob/master/packages/exo/docs/exo-taxonomy.md#heap-vs-virtual-vs-durable)):
 
-* singleton in-RAM objects (created with `Far()` from a standard JavaScript
-  `Object`, using the standard objects-as-closures pattern)
+* Heap objects in vat RAM
+  * one-off objects created with `Far()` or `makeExo()`
+  * instances created with a "make" function from `defineExoClass()`
+  * multifaceted kit instances created with a "makeKit" function from `defineExoClassKit()`
 
-* virtual objects (defined with `makeKind()` or `makeDurableKind()`, created
-  by invoking the function those return, and whose state is kept in the
-  database rather than in RAM)
+* Virtual objects in disk-based storage
+  * instances created with a "make" function from `defineVirtualExoClass()`
+  * multifaceted kit instances created with a "makeKit" function from `defineVirtualExoClassKit()`
 
-The v2 code is obligated to re-define all durable virtual-object Kinds created by v1 during the upgrade phase. Once complete, this allows liveslots to satisfy deserialization of any inbound message addressed to (or referencing) a previously-exported virtual object.
+* Durable objects in disk-based storage
+  * instances created with a "make" function from `defineDurableExoClass()` or `prepareExoClass()`
+  * multifaceted kit instances created with a "makeKit" function from `defineDurableExoClassKit()` or `prepareExoClassKit()`
+  * singleton objects created with `prepareExo()`
 
-The v2 code is also obligated to reattach any singleton objects. The API for this is still under consideration. The root object may be a special case: if the upgrade invocation is simply a new call to `buildRootObject()`, then whatever object that returns will be associated with the root-object vref (`o+0`).
+During the upgrade phase, v2 code is obligated to re-define all durable Kinds created by v1 (i.e., those associated with objects in the third category). Once complete, this allows liveslots to satisfy deserialization of any inbound message addressed to (or referencing) a previously-exported durable object.
```

Comment:
Worth mentioning: The kinds have to have the same set of named `facets` as they did originally.